hermes/bin/src/runtime/host/hermes/hash/mod.rs

```diff
@@ -23,18 +25,36 @@ impl Stateful for State {
 impl Host for HermesState {
     /// Hash a binary buffer with BLAKE2s
     fn blake2s(
-        &mut self, _buf: Bstr, _outlen: Option<u8>, _key: Option<Bstr>,
+        &mut self, _buf: Bstr, _outlen: Option<u8>,
     ) -> wasmtime::Result<Result<Bstr, Errno>> {
         todo!()
     }
 
-    /// Hash a binary buffer with `BLAKE2b`
-    fn blake2b(
-        &mut self, _buf: Bstr, _outlen: Option<u8>, _key: Option<Bstr>,
+    /// Hash a binary buffer with `BLAKE2s` with `MAC` (Message Authentication Code) mode
+    fn blake2smac(
+        &mut self, _buf: Bstr, _outlen: Option<u8>, _key: Bstr, _salt: Option<Bstr>,
+        _persona: Option<Bstr>,
     ) -> wasmtime::Result<Result<Bstr, Errno>> {
         todo!()
     }
 
+    /// Hash a binary buffer with `BLAKE2b`
+    fn blake2b(&mut self, buf: Bstr, outlen: Option<u8>) -> wasmtime::Result<Result<Bstr, Errno>> {
+        blake2b::blake2b_impl(&buf, outlen)
```

Comment:
I am not sure that this is a correct conversion which you are doing here.
It seems like you are converting a `Errno` to the `wasmtime::Error` which is not correct for this place.
I think here you should return the following.
```
Ok( blake2b::blake2b_impl(&buf, outlen))
```